metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/service/zoho/ZohoAccessService.java

```diff
@@ -0,0 +1,230 @@
+package eu.europeana.enrichment.service.zoho;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import eu.europeana.corelib.definitions.edm.entity.Organization;
+import eu.europeana.corelib.solr.entity.Address;
+import eu.europeana.corelib.solr.entity.OrganizationImpl;
+import eu.europeana.enrichment.service.exception.ZohoAccessException;
+import eu.europeana.enrichment.service.zoho.model.ZohoOrganizationAdapter;
+import eu.europeana.metis.authentication.dao.PsqlMetisUserDao;
+import eu.europeana.metis.authentication.dao.ZohoAccessClientDao;
+import eu.europeana.metis.authentication.dao.ZohoFields;
+import eu.europeana.metis.exception.GenericMetisException;
+
+/**
+ * @author Simon Tzanakis (Simon.Tzanakis@europeana.eu)
+ * @since 2017-10-27
+ */
+@Service
+public class ZohoAccessService {
```

Comment:
More and more reading this class I think it does Adapter specific work.
Perhaps the structure could be a ZohoOrganization and a JsonNodeToZohoOrganizationAdapter class that handles all the code about JsonNode and ZohoOrganization(see other comments about the ZohoOrganizationAdapter). And another adapter that would create an Organization from a ZohoOrganization?
In that way the ZohoAccessClientDao doesn't even need to be there, wherever the script would be created then there the ZohoAccessClientDao can get the JsonNodes and use all the above to structure everything. This way I think tests are also more maintainable.
What do you think?

Reply:
actually the ZohoOrganizationAdapter is a wrapper class. It would be ok to change the name. 
The ZohoOrganization is not fully compatible with corelib Organization. Data conversions are needed, which I consider to belong to business logic (service).  As we don't have complete specifications and we don't know the differences from other sources, like wikidata, I would like to keep it simple for now.  I would also not like to add business logic to Dao level. I would like to keep it as simple/dummy as possible, but I would keep the ZohoAccessDao to be the only class running the Zoho API invocation (http requests)